### PR TITLE
Issue 950: samplesheet validation - no spaces in sample or patient names

### LINKED ISSUE #950: More check for the samplesheet
### Description of feature

no space in sample or id

### PR DESCRIPTION
**Pull request by matrulda**
This fixes #950 

Note that the pipeline is terminated using System.exit() which isn't recommended (https://github.com/nf-core/tools/issues/1840). But I leave that for another PR since it has to be changed in multiple places.

Tested successfully (locally) with: 
```
TMPDIR=~ PROFILE=docker pytest --tag sample_with_space  --symlink --kwdof --git-aware --color=yes
```

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: tests/test_samplesheet_validation_spaces.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- name: Test that pipeline fail when sample/patient name contain space
+  command: nextflow run main.nf -profile test --input ./tests/csv/3.0/sample_with_space.csv --outdir results
+  tags:
+    - sample_with_space
```
Comment (maxulysse) [inline tests/test_samplesheet_validation_spaces.yml:4]:
```suggestion
    - sample_with_space
    - validation_checks
```

**Comment by maxulysse** [comment]
Can you add this bit in `tests/config/tags.yml`?
```
validation_checks:
  - conf/modules/**
  - main.nf
  - modules/**
  - nextflow.config
  - nextflow_schema.json
  - subworkflows/**
  - tests/test_skip_markduplicates.yml
  - workflows/**
```

**Comment by maxulysse** [comment]
And also this part in `.github/workflows/pytest-workflow.yml` in the exclude part (around L41-L57)
```
          - profile: "singularity"
            tags: validation_checks
          - profile: "conda"
            tags: validation_checks
```

**Comment by matrulda** [comment]
Thanks @maxulysse ! I think I fixed it all now :)

**Review by maxulysse — APPROVED** [review]

**Pull request merged**